_docs/Qabel-Register.md

```diff
@@ -0,0 +1,113 @@
+---
+title: "Identity directory"
+---
+# Identity directory
+
+## Abstract
+
+Identities can be publicly shared in an identity directory provided by the
+register server.
+
+All data is sent as JSON (encoded in UTF-8). All data types are defined
+[here](../Qabel-Client-Local-Data#data-types).
+
+## Concept & structures
+
+The identity directory can be queried using private data (*fields*),
+e.g. an email address or phone number. The register server then
+returns matching identites to the requester.
+
+Changes to the directory are submitted as *update requests*. An update
+request can associate or de-associate any number of fields with/from
+an identity (the combination of a key-pair, a drop URL and an
+alias). The association of a particular field with an identity is an
+*entry* in the directory. Entries are not unique.
+
+An identity is always represented using this JSON structure:
+
+    {
+        "public_key": "hex of 32-byte public key",
+        "drop_url": "drop protocol URL",
+        "alias": "user-specified alias"
+    }
+
+### Fields
+
+Currently defined fields are:
+
+email
+: E-Mail address
+
+phone
+: Phone number (must be capable of receiving SMS)
+
+## APIs
+
+### Search
+
+* Resource: /api/v0/search/
+* Method: GET
+* Request data: field-value pairs (`field=value[&field=value...]`)
+* Response data: `[identity, ...]`
+
+When multiple field-value pairs are specified only identities matching
+all pairs will be returned.
+
+At least one field-value pair must be specified.
+
+### Update
+
+Atomically create or delete entries (therefore also update entries).
+
+* Resource: /api/v0/update/
+* Method: PUT
+* Content types:
+
+    * application/json
+    * application/vnd.qabel.noisebox+json
+
+    If the content type is plain text it contains an *update
+    request*. This is only valid for requests purely made of
+    *deletes*.
+
+    If the content is a noise box, then that noise box must be
+    encrypted for the servers [ephemeral key](#key) and it's signature
+    must be made by the key pair the update request refers to.
```

Comment:
The noise box normally does not contain a signature. If only the noise box is used to proof the identity (which is the status I know), this proof is done by computing the DH, which is required for creating a noise box. Hence if the noise box can be decrypted by the server and the creators key equals the key of the modified identity, the proof is valid.
Thus the sentence should read as follows: "..., then that noise box must be created by the key pair the update request refers to and it must be encrypted for the servers [ephemeral key](#key)." Additionally this could be explained by writing: "A valid noise box can only be computed by calculating the Diffie-Hellman of the requesting key pair and the servers ephemeral key. Forging a noise box would be as hard as breaking the CDH-Problem".